### PR TITLE
G-CM-35 | Refactor CM to POST consent-artefacts to HIP

### PR DESCRIPTION
**Pull request by Saisree9**

### CODE REVIEW

File: src/main/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifier.java [hunk 1]
```diff
@@ -40,4 +52,26 @@
...
+                                .post()
+                                .uri(gatewayServiceProperties.getBaseUrl() + HIP_CONSENT_NOTIFICATION_URL_PATH)
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .header(HttpHeaders.AUTHORIZATION, token)
+                                .header(HDR_HIP_ID, hipId)
+                                .bodyValue(body)
+                                .retrieve()
+                                .onStatus(httpStatus -> httpStatus.value() == 401,
+                                        // Error msg should be logged
+                                        clientResponse -> Mono.error(ClientError.unAuthorized()))
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifier.java:69]: you can log it now, don't need to be a separate commit :)

File: src/main/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifier.java [hunk 2]
```diff
@@ -16,6 +20,10 @@
 public class ConsentArtefactNotifier {
     private final WebClient.Builder webClientBuilder;
     private final Supplier<Mono<String>> tokenGenerator;
+    private final GatewayServiceProperties gatewayServiceProperties;
+
+    private static final String HDR_HIP_ID = "X-HIP-ID";
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifier.java:25]: Check whether it's used in multiple places, please create a consent file and use it everywhere.

File: src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java [hunk 3]
```diff
@@ -50,7 +54,7 @@ public void subscribe() throws ClientError {
                 logger.info("Received notify consent to hip for consent artefact: {}",
                         consentArtefact.getConsentId());
 
-                sendConsentArtefact(consentArtefact);
+                sendConsentArtefactToHIP(consentArtefact);
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java:57]: This call will break the existing implementation right? if gateway and HIP are not ready when you merge this code?

Comment (Saisree9) [reply src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java:57]: @ganesanarun Yes.

Comment (ganesanarun) [reply src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java:57]: How do we avoid that? do you have an idea?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5637360&bid=18462397)

File: src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java (hunk 3 above)
Comment (Saisree9) [reply src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java:57]: @ganesanarun Maybe, can we create a new listener for the same queue without disturbing the HIPConsentNotificationListener?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  1
- src/main/java/in/projecteka/consentmanager/clients/HeaderConstants.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5637360&bid=18462397)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/clients/HeaderConstants.java  1
- src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5637360&bid=18462397)

File: src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java [hunk 4]
```diff
@@ -50,7 +54,7 @@ public void subscribe() throws ClientError {
                 logger.info("Received notify consent to hip for consent artefact: {}",
                         consentArtefact.getConsentId());
 
-                sendConsentArtefact(consentArtefact);
+                sendConsentArtefactToHIP(consentArtefact).block();
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/consent/HipConsentNotificationListener.java:57]: there was no block? now there's? is it really needed?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/clients/HeaderConstants.java  1
- src/test/java/in/projecteka/consentmanager/clients/ConsentArtefactNotifierTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5637360&bid=18462397)
